orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -51,14 +54,16 @@
   private final boolean caseSensitive;
   private final Function<TypeDescription, OrcBatchReader<?>> batchReaderFunction;
   private final int recordsPerBatch;
+  private final NameMapping nameMapping;
 
   OrcIterable(InputFile file, Configuration config, Schema schema,
-              Long start, Long length,
+              NameMapping nameMapping, Long start, Long length,
               Function<TypeDescription, OrcRowReader<?>> readerFunction, boolean caseSensitive, Expression filter,
               Function<TypeDescription, OrcBatchReader<?>> batchReaderFunction, int recordsPerBatch) {
     this.schema = schema;
     this.readerFunction = readerFunction;
     this.file = file;
+    this.nameMapping = Optional.ofNullable(nameMapping).orElse(MappingUtil.create(schema));
```

Comment:
Should we also check if the file schema does not have ids before initializing nameMapping?

Reply:
Done.